smtk/mesh/Collection.cxx

```diff
@@ -502,5 +502,33 @@ bool Collection::setNeumannOnMeshes(const smtk::mesh::MeshSet& meshes,
   return false;
 }
 
+//----------------------------------------------------------------------------
+bool Collection::associateModel(const smtk::common::UUID& uuid)
+{
+  if(this->m_modelEntity == uuid)
+    {
+    return true;
+    }
+
+  if(!this->m_modelEntity.isNull() && uuid.isNull())
+    {
+    // TODO: should we clear everything and make this an invalid collection (how?)
+    return false;
+    }
+
+  smtk::model::Model inModel(this->m_modelManager.lock(), uuid);
+  // TODO: what if this->m_modelEntity is valid, should we clear everything (how?)
+  if(inModel.isValid() && this->m_modelEntity.isNull())
+    {
+    this->m_modelEntity = uuid;
```

Comment:
I wouldn't store m_modelEntity, instead use getModelEntity() on the mesh_iface to get the uuid.